jest/preprocessor.js

```diff
@@ -21,8 +23,16 @@ module.exports = {
       if (isJISON) {
         src = new jison.Generator(src).generate();
       }
+      if (filename.endsWith(".ts") || filename.endsWith(".tsx")) {
```

Comment:
Is there anything that holds us back from using a separate transform statement? [It should be possible like this](https://github.com/mesosphere/mockserver/blob/master/package.json#L58).

For me this is a lot cleaner and I would like to move away from this file as it has some more complexity

Reply:
If it's about our babel file not being picked up, they have docs on that: https://github.com/kulshekhar/ts-jest#using-a-custom-babel-config

Reply:
As discussed: I decided to extract jest config refactoring from this piece of work.